## Assignment #1 DATA 730

This assignment tests that you have the course tech stack set up. If you do, you will be able to complete this notebook and turn-in your work by pushing to your assignment repository.

## Tech stack checklist

*Fill out the checklist below once you have completed each task:*

- [ ] GitHub account

Change the cell below 👇 to a markdown cell and write in your GitHub account handle

noorasfour09



In [8]:
# uncomment the following line and execute this cell to demonstrate that you have installed the tidymodels package
library('tidymodels')

## Let's do some statistical inference!

Read [this article](https://www.theguardian.com/world/2002/jan/04/euro.eu2). 

Execute the code below to investigate whether the coin is fair...

In [9]:
# These functions simulate tossing a fair coin n_tosses number of times for n_iterations number of iterations

toss_fair_coin_n_times = function(n_tosses = 250) {
    result = ifelse(runif(n = n_tosses) < 0.5, 'heads', 'tails')
    result
}

repeat_the_experiment = function(n_iterations, n_tosses = 250) {
    n_heads_each_iteration = c()
    for (i in 1:n_iterations) {
        n_heads = sum(toss_fair_coin_n_times(n_tosses) == 'heads')
        n_heads_each_iteration = c(n_heads_each_iteration, n_heads)
    } 
    n_heads_each_iteration
}

In [11]:
# Let's do 10,000 experiments of 250 tosses and see how often our outcome deviates from what was actually observed (139 heads)

n_iterations = 10000
n_tosses = 250
observed_deviation = 139 - 125 # 139 heads observed, expected value for a fair coin of 125

n_heads_each_experiment = repeat_the_experiment(n_iterations, n_tosses)

x = mean(abs(n_heads_each_experiment - 125) >= observed_deviation)
paste0('With a fair coin we would expect a deviation as large or larger than what we observed ', x*100, '% of times')

[1] "With a fair coin we would expect a deviation as large or larger than what we observed 8.94% of times"

**Do you think the coin was fair?** (Answer below in a markdown cell.)

No, I don't think we have strong enough evidence to say the coin is unfair. The code ran 10,000 iterations of this experiment and the deviation we observed happened 8.94% of the time. I dont't think that evidence is significant enough to prove the coin unfair. 